### PR TITLE
Add Focus component to focus an element on receiving an event

### PR DESCRIPTION
**Pull request by ifitzpatrick**
Component for responding to events by focusing an element. Focus won't work when responding to certain events without being wrapped by window.setTimeout as far as I know, but if there is a better way to do that I would love to know.

### CODE REVIEW

**Comment by bergie** [comment]
@ifitzpatrick I just updated the build setup, can you rebase this on master?

**Comment by ifitzpatrick** [comment]
opening a new pr rebased on master

File: spec/runner.html [hunk 1]
```diff
@@ -39,8 +39,13 @@
     <script src="./ListenMouse.js"></script>
     <script src="./ListenScroll.js"></script>
     <script src="./ListenTouch.js"></script>
+    <script src="./Focus.js"></script>
     <script>
-      var runner = mocha.run();
+      if (window.mochaPhantomJS) {
```
Comment (ifitzpatrick) [inline spec/runner.html]: Fixes http://stackoverflow.com/questions/15657458/mocha-init-timeout-with-mocha-phantomjs

Comment (bergie) [inline spec/runner.html]: This should not be necessary with mocha-phantomjs 3.0

Comment (ifitzpatrick) [inline spec/runner.html]: Sorry didn't update my dependencies when I rebased. Removing this commit.

**Comment by bergie** [comment]
Thanks!

**Pull request merged**